music5000.js

```diff
@@ -0,0 +1,195 @@
+"use strict";
+
+// Code ported from Beebem (C to .js) by Jason Robson
+const AUDIO_BUFFER_SIZE = 256;
+
+const RAM_SIZE = 2048;
+const WAVE_TABLE_SIZE = 128;
+const WAVE_TABLES = 14;
+const NUM_CHANNELS = 16;
+const CHANNEL_REG_OFFSET = WAVE_TABLES * WAVE_TABLE_SIZE;
+const CHANNEL_ROW_SIZE = 128;
+
+// Control register bits
+const CTRL_STEREO_POS = (0xf);
+const CTRL_INVERT_WAVE = (1 << 4);
+const CTRL_MODULATE_ADJ = (1 << 5);
+
+// Data bits
+const DATA_SIGN = (1 << 7);
+const DATA_VALUE = (0x7f);
+const FREQ_DISABLE = (1 << 0);
+const NOT_FREQ_DISABLE = 0xfe;
+
+// Channel register sets
+const REG_SET_NORMAL = 0;
+const REG_SET_ALT = 1;
+
+
+export function Music5000(onBuffer) {
```

Comment:
For what it's worth I'm now moving more things to newer JS versions - you can use `class` here instead of the old-style function-based classes. Not a problem either way, and most of the code still uses the older style, so it fits right in :)

Reply:
I can switch these over to classes if you would prefer? I did it like this purely to match the rest of the code I was seeing. Happy either way...

Reply:
Whichever you fancy :) I don't want to force you through hoops; so thisi s fine as-is, but I'll (sloowwwwwly) be moving towards JS classes in the rest of the codebase.